kong/pdk/client.lua

```diff
@@ -241,6 +242,49 @@ local function new(self)
     return is_tls and "tls" or "tcp"
   end
 
+  ---
+  -- Returns a table of `authenticated_groups` of the currently authenticated
+  -- consumer, or nil if groups is not set.
+  -- @function kong.client.get_authenticated_groups
+  -- @phases access, header_filter, body_filter, log
+  -- @treturn table|nil
+  -- The table will have an array part to iterate over, and a hash part where
+  -- each group name is indexed by itself. Eg.
+  -- {
+  --   [1] = "users",
+  --   [2] = "admins",
+  --   users = "users",
+  --   admins = "admins",
+  -- }
+  -- @usage
+  -- local groups = kong.client.get_authenticated_groups()
+  -- if not groups then
+  --   for i = 1, #groups_to_check do
+  --     if groups[i] == 'group1' then
+  --       return true
+  --     end
+  --   end
+  -- end
+  function _CLIENT.get_authenticated_groups()
+    check_phase(AUTH_AND_LATER)
+
+    local authenticated_groups = ngx.ctx.authenticated_groups
+    if authenticated_groups == nil then
+      return nil
+    end
+
+    assert(type(authenticated_groups) == "table",
+           "invalid authenticated_groups, a table was expected")
```

Comment:
Please don't use `assert`, but use `error(..., 2)` (with the proper stacktrace level) instead. Assert will not report the stacktrace from the caller's stack level.

Reply:
I think assert is fine here. The error condition is not the users fault, it's a bug elsewhere, and in that case you'll want the full trace.

If a bad parameter is passed in to the function, then doing an error at level 2 would make sense